kafka-client-lib/src/main/java/io/jenkins/plugins/remotingkafka/KafkaUtils.java

```diff
@@ -20,28 +20,37 @@
 
 import java.util.ArrayList;
 import java.util.Properties;
+import java.util.logging.Logger;
 
 public class KafkaUtils {
-    public static Producer<String, byte[]> createByteProducer(String kafkaURL) throws RemotingKafkaConfigurationException {
+    private static final Logger LOGGER = Logger.getLogger(KafkaUtils.class.getName());
+
+    public static Producer<String, byte[]> createByteProducer(String kafkaURL, Properties securityProps)
```

Comment:
I would recommend to keep `securityProps ` nullable so that API users can easily implement non-authenticated mode. If you agree, I would use the `@CheckForNull` annotation in APIs so that it's handled correctly in the code